docs/user_guide/secure_payloads.rst

```diff
@@ -0,0 +1,233 @@
+to get an idea of how to write your own.  See the files starting with `local_action`
+in https://github.com/keylime/keylime/tree/master/auto-ipsec/libreswan/src
+
+There are some conventions to specifying revocation actions. As described above,
+their names must start with `local_action` to be executed. They also must be
+listed (without `.py` extensions) in a comma separated list in a file called
+`action_list` in the zip file.  For example to run `local_action_a.py` and
+`local_action_b.py` the `action_list` file should contain `local_action_a,local_action_b`.
+
+So far we've described all the details of this in fine detail, but much of this
+automation will happen by default.
+
+Certificate Package Example
+---------------------------
+
+Let's put all of the above together with an example.
+
+For the following example, we will provision some SSH keys onto the Agent.
+
+1. Create a directory to host the files and `autorun.sh` script. For this example, we will use the directory `payload`
+2. Create an `autorun.sh` script in the `payload` directory:
+
+```
+#!/bin/bash
+
+# this will make it easier for us to find our own cert
+ln -s `ls *-cert.crt | grep -v Revocation` mycert.crt
+
+mkdir -p /root/.ssh/
+cp payload_id_rsa* /root/.ssh/
+chmod 600 /root/.ssh/payload_id_rsa*
+```
+
+3. Copy the files you wish to provision into the `payload` directory.
+
+```
+$ ls payload/
+autorun.sh
+payload_id_rsa.pub
+payload_id_rsa
+```
+
+Send the files using the Keylime Tenant tool:
+
+```
+keylime_tenant -t <agent-ip> --cert myca --include payload
+```
+
+Recall that the `--cert` option tells Keylime to create a certificate authority
+at the default location `/var/lib/keylime/ca` and give this machine an X509
+identity with its UUID. Keylime will also create a revocation notifier
+certificate for this CA and make it available to the verifier. Finally, the
+`--include` option tells Keylime to securely deliver the files in the specified
+directory (`payload` in our case) along with the X509 certs to the targeted
+Agent machine.
+
+If the enrolment was been successful, you will be able to see the contents of
+the `payload` directory in `/var/lib/keylime/secure/unzipped` along with the
+certs and included files. You should also see the SSH keys we included made in
+`/root/.ssh` directory from where the autorun.sh script was ran.
+
+Now, let's extend this example with revocation.  In this example, we're going to
+execute a simple revocation action on the node that was revoked.
+
+It is also possible to configure scripts for execution should a node fail any
+given criteria (IMA measurements, for example).
+
+To configure this, we will use our `payload` directory again.
+
+First create a Python script with the preface of `local_action`
+
+For example `local_action_rm_ssh.py`
+
+Within this script create an `execute` function:
+
+```
+import os
+from M2Crypto import X509
+import keylime.secure_mount as secure_mount
+
+async def execute(json_revocation):
```

Comment:
It might be better to name this variable something like `event` vs `json_revocation`. Because at this point it's not JSON (even if it started that way before it got to us) and it also might not be a revocation (hence the check on line 200).